test/common/testlib.py

```diff
@@ -490,38 +497,45 @@ def run(self, result=None):
             if stopTestRun is not None:
                 stopTestRun()
 
-    def setUp(self, macaddr=None, memory_mb=None, cpus=None):
+    def setUp(self):
+        if opts.address and self.provision is not None:
+            raise unittest.SkipTest("Cannot provision multiple machines if a specific machine address is specified")
```

Comment:
If this behavior is to be changed it should be done as a follow up. It only affects running of tests manually against a running system.